From e50e06362f8faa8033d5169390aa5d9d2d7e796f Mon Sep 17 00:00:00 2001
From: Tom Rondeau <tom@trondeau.com>
Date: Tue, 18 Mar 2014 09:56:33 -0400
Subject: [PATCH 01/13] qtgui: fixing missing include file.

---
 gr-qtgui/lib/qtgui_util.cc |    1 +
 1 file changed, 1 insertion(+)

diff --git a/gr-qtgui/lib/qtgui_util.cc b/gr-qtgui/lib/qtgui_util.cc
index 9ca19cb..07cf3cc 100644
--- a/gr-qtgui/lib/qtgui_util.cc
+++ b/gr-qtgui/lib/qtgui_util.cc
@@ -22,6 +22,7 @@
 
 #include <gnuradio/qtgui/utils.h>
 #include <QDebug>
+#include <QFile>
 
 QString
 get_qt_style_sheet(QString filename)
-- 
1.7.10.4

