From 26475d40e123a139cd7029c5172612d6f59d9fd3 Mon Sep 17 00:00:00 2001
From: Johnathan Corgan <johnathan@corganlabs.com>
Date: Fri, 3 Mar 2017 08:02:35 -0800
Subject: [PATCH 01/72] Updated revision to 3.7.11.1

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0ac65feda5..650cfe3ac4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -47,7 +47,7 @@ message(STATUS "Build type set to ${CMAKE_BUILD_TYPE}.")
 set(VERSION_INFO_MAJOR_VERSION 3)
 set(VERSION_INFO_API_COMPAT    7)
 set(VERSION_INFO_MINOR_VERSION 11)
-set(VERSION_INFO_MAINT_VERSION 0)
+set(VERSION_INFO_MAINT_VERSION 1)
 include(GrVersion) #setup version info
 
 # Append -O2 optimization flag for Debug builds (Not on MSVC since conflicts with RTC1 flag)
-- 
2.11.0

