From 20ef863e86289107bc4c6cc375edd5890a363694 Mon Sep 17 00:00:00 2001
From: Sebastian Koslowski <sebastian.koslowski@gmail.com>
Date: Wed, 4 Apr 2018 23:28:19 +0200
Subject: [PATCH 2/7] grc: fix param tooltip error (don't use string/$string)

---
 grc/gui/Param.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/grc/gui/Param.py b/grc/gui/Param.py
index 7f90a7bea1..019bc533e7 100644
--- a/grc/gui/Param.py
+++ b/grc/gui/Param.py
@@ -368,9 +368,9 @@ PARAM_LABEL_MARKUP_TMPL="""\
 
 TIP_MARKUP_TMPL="""\
 ########################################
-#def truncate(string)
+#def truncate(value)
     #set $max_len = 100
-    #set $string = str($string)
+    #set $string = str(value)
     #if len($string) > $max_len
 $('%s...%s'%($string[:$max_len/2], $string[-$max_len/2:]))#slurp
     #else
-- 
2.11.0

