From edb12325a82b11fc7cc3ab9572dbcab3d8f378a7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marcus=20M=C3=BCller?= <marcus.mueller@ettus.com>
Date: Tue, 21 Mar 2017 17:52:39 +0100
Subject: [PATCH 08/72] added the missing == operator.

---
 gr-uhd/swig/uhd_swig.i | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gr-uhd/swig/uhd_swig.i b/gr-uhd/swig/uhd_swig.i
index b82d0fdae9..2852447c80 100644
--- a/gr-uhd/swig/uhd_swig.i
+++ b/gr-uhd/swig/uhd_swig.i
@@ -105,6 +105,10 @@
         temp -= what;
         return temp;
     }
+    bool __eq__(const uhd::time_spec_t &what)
+    {
+      return (what == *self);
+    }
 };
 
 %include <uhd/types/stream_cmd.hpp>
-- 
2.11.0

