From c62ad7565afaed3887cd1161cd79ae60ed4a8383 Mon Sep 17 00:00:00 2001
From: Sebastian Koslowski <koslowski@kit.edu>
Date: Wed, 29 Mar 2017 09:15:46 +0200
Subject: [PATCH 11/72] docs: nitpicky minor fix

---
 docs/doxygen/other/pmt.dox | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/docs/doxygen/other/pmt.dox b/docs/doxygen/other/pmt.dox
index 76f598809d..932f6c0a83 100644
--- a/docs/doxygen/other/pmt.dox
+++ b/docs/doxygen/other/pmt.dox
@@ -83,7 +83,7 @@ std::string str = pmt::symbol_to_string(P_str);
 
 The pmt::intern is another way of saying pmt::string_to_symbol.
 
-In Python, we can make use of the weak typing, and there's actually a
+In Python, we can make use of the dynamic typing, and there's actually a
 helper function to do these conversions (C++ also has a helper
 function for converting to PMTs called pmt::mp(), but its less
 powerful, and not quite as useful, because types are always strictly
-- 
2.11.0

