From 641335d52f850084735278cbfeca9091190cb4cd Mon Sep 17 00:00:00 2001
From: Tom Rondeau <tom@trondeau.com>
Date: Fri, 6 Sep 2013 11:58:14 -0400
Subject: [PATCH 12/40] runtime: exporting msg_edge_list now, too.

---
 gnuradio-runtime/swig/top_block.i |    1 +
 1 file changed, 1 insertion(+)

diff --git a/gnuradio-runtime/swig/top_block.i b/gnuradio-runtime/swig/top_block.i
index f5c5819..4851599 100644
--- a/gnuradio-runtime/swig/top_block.i
+++ b/gnuradio-runtime/swig/top_block.i
@@ -44,6 +44,7 @@ namespace gr {
     void lock();
     void unlock() throw (std::runtime_error);
     std::string edge_list();
+    std::string msg_edge_list();
     void dump();
 
     int max_noutput_items();
-- 
1.7.10.4

