From 0471d1f5c32926322df463ebf585e4bd4a9b2864 Mon Sep 17 00:00:00 2001
From: Sebastian Koslowski <koslowski@kit.edu>
Date: Tue, 12 Apr 2016 12:38:03 +0200
Subject: [PATCH 13/28] grc: update year in copyright

---
 grc/python/Platform.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/grc/python/Platform.py b/grc/python/Platform.py
index 1d93276..2a0bbf9 100644
--- a/grc/python/Platform.py
+++ b/grc/python/Platform.py
@@ -1,5 +1,5 @@
 """
-Copyright 2008-2015 Free Software Foundation, Inc.
+Copyright 2008-2016 Free Software Foundation, Inc.
 This file is part of GNU Radio
 
 GNU Radio Companion is free software; you can redistribute it and/or
-- 
2.1.4

