From cce68f26f1641c0a97b0bbc9a4608903aed493c7 Mon Sep 17 00:00:00 2001
From: Bastian Bloessl <bloessl@ccs-labs.org>
Date: Fri, 30 Sep 2016 06:52:22 +0200
Subject: [PATCH 18/20] runtime: connect message ports after unlock()

---
 gnuradio-runtime/lib/flat_flowgraph.cc | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/gnuradio-runtime/lib/flat_flowgraph.cc b/gnuradio-runtime/lib/flat_flowgraph.cc
index 434a92f..b7e9491 100644
--- a/gnuradio-runtime/lib/flat_flowgraph.cc
+++ b/gnuradio-runtime/lib/flat_flowgraph.cc
@@ -316,6 +316,15 @@ namespace gr {
         setup_buffer_alignment(block);
       }
 
+      // Connect message ports connetions
+      for(msg_edge_viter_t i = d_msg_edges.begin(); i != d_msg_edges.end(); i++) {
+          if(FLAT_FLOWGRAPH_DEBUG)
+              std::cout << boost::format("flat_fg connecting msg primitives: (%s, %s)->(%s, %s)\n") %
+                  i->src().block() % i->src().port() %
+                  i->dst().block() % i->dst().port();
+          i->src().block()->message_port_sub(i->src().port(), pmt::cons(i->dst().block()->alias_pmt(), i->dst().port()));
+      }
+
       // Now deal with the fact that the block details might have
       // changed numbers of inputs and outputs vs. in the old
       // flowgraph.
-- 
2.1.4

