From a3e22a162a9a567fc4159635bbb09f3a77c0fba7 Mon Sep 17 00:00:00 2001
From: Sebastian Koslowski <koslowski@kit.edu>
Date: Mon, 25 Apr 2016 11:28:12 +0200
Subject: [PATCH 19/28] grc: show underscores in recent files menu

---
 grc/gui/Bars.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/grc/gui/Bars.py b/grc/gui/Bars.py
index 19f041f..8b389a7 100644
--- a/grc/gui/Bars.py
+++ b/grc/gui/Bars.py
@@ -188,7 +188,7 @@ class SubMenuCreator(object):
                 item = Actions.FLOW_GRAPH_NEW.create_menu_item()
                 item.set_label(name)
             else:
-                item = gtk.MenuItem(name)
+                item = gtk.MenuItem(name, use_underline=False)
                 item.connect('activate', self.callback_adaptor, (action, key))
             menu.append(item)
         menu.show_all()
@@ -201,7 +201,7 @@ class SubMenuCreator(object):
         recent_files = Preferences.get_recent_files()
         if len(recent_files) > 0:
             for i, file_name in enumerate(recent_files):
-                item = gtk.MenuItem("%d. %s" % (i+1, file_name))
+                item = gtk.MenuItem("%d. %s" % (i+1, file_name), use_underline=False)
                 item.connect('activate', self.callback_adaptor,
                              (action, file_name))
                 menu.append(item)
-- 
2.1.4

