From 89af584a24f470027874e30a2e238de3bb5f0d1f Mon Sep 17 00:00:00 2001
From: Josh Blum <josh@joshknows.com>
Date: Thu, 2 Feb 2012 13:25:50 -0800
Subject: [PATCH 19/47] grc: fix install permissions on grc_setup_freedesktop

---
 grc/freedesktop/CMakeLists.txt |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/grc/freedesktop/CMakeLists.txt b/grc/freedesktop/CMakeLists.txt
index 8c0f53d..492b0f7 100644
--- a/grc/freedesktop/CMakeLists.txt
+++ b/grc/freedesktop/CMakeLists.txt
@@ -1,4 +1,4 @@
-# Copyright 2011 Free Software Foundation, Inc.
+# Copyright 2011-2012 Free Software Foundation, Inc.
 # 
 # This file is part of GNU Radio
 # 
@@ -41,7 +41,7 @@ if(UNIX AND HAVE_XDG_UTILS)
         ${CMAKE_CURRENT_BINARY_DIR}/grc_setup_freedesktop
     @ONLY)
     install(
-        FILES ${CMAKE_CURRENT_BINARY_DIR}/grc_setup_freedesktop
+        PROGRAMS ${CMAKE_CURRENT_BINARY_DIR}/grc_setup_freedesktop
         DESTINATION ${GR_PKG_LIBEXEC_DIR} COMPONENT "grc"
     )
 endif(UNIX AND HAVE_XDG_UTILS)
-- 
1.7.9.1

