From 63ffe1ade4b582fc76a09139231dbf942c6ec85a Mon Sep 17 00:00:00 2001
From: Josh Blum <josh@joshknows.com>
Date: Thu, 2 Feb 2012 13:30:34 -0800
Subject: [PATCH 20/47] grc: better error for import gr fail

---
 grc/scripts/gnuradio-companion |   13 +++++++++++--
 1 files changed, 11 insertions(+), 2 deletions(-)

diff --git a/grc/scripts/gnuradio-companion b/grc/scripts/gnuradio-companion
index a4115c3..e76322b 100755
--- a/grc/scripts/gnuradio-companion
+++ b/grc/scripts/gnuradio-companion
@@ -1,6 +1,6 @@
 #!/usr/bin/env python
 """
-Copyright 2009 Free Software Foundation, Inc.
+Copyright 2009-2012 Free Software Foundation, Inc.
 This file is part of GNU Radio
 
 GNU Radio Companion is free software; you can redistribute it and/or
@@ -25,7 +25,16 @@ import gtk
 try: from gnuradio import gr
 except ImportError, e:
 	d = gtk.MessageDialog(type=gtk.MESSAGE_ERROR, buttons=gtk.BUTTONS_CLOSE, message_format="""
-Cannot import gnuradio. Are your PYTHONPATH and LD_LIBRARY_PATH set correctly?""")
+Cannot import gnuradio.
+
+Is the python path environment variable set correctly?
+	All OS: PYTHONPATH
+
+Is the library path environment variable set correctly?
+	Linux: LD_LIBRARY_PATH
+	Windows: PATH
+	MacOSX: DYLD_LIBRARY_PATH
+""")
 	d.set_title(str(e))
 	d.run()
 	exit(-1)
-- 
1.7.9.1

