From c4a7a9a620fc9a1117b70e463b071ab4be9539a2 Mon Sep 17 00:00:00 2001
From: Sebastian Koslowski <koslowski@kit.edu>
Date: Mon, 25 Apr 2016 11:29:01 +0200
Subject: [PATCH 20/28] grc: don't try to auto-generate non-hier blocks

---
 grc/python/Platform.py | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/grc/python/Platform.py b/grc/python/Platform.py
index 2a0bbf9..351f04c 100644
--- a/grc/python/Platform.py
+++ b/grc/python/Platform.py
@@ -146,6 +146,8 @@ class Platform(_Platform, _GUIPlatform):
             flow_graph.validate()
             if not flow_graph.is_valid():
                 raise Exception('Flowgraph invalid')
+            if not flow_graph.get_option('generate_options').startswith('hb'):
+                raise Exception('Not a hier block')
         except Exception as e:
             Messages.send('>>> Load Error: %r: %s\n' % (file_path, str(e)))
             return False
-- 
2.1.4

