From 2b87c2a7c11c4ee20b8f1356f39537043234c1b4 Mon Sep 17 00:00:00 2001
From: Nicholas Corgan <n.corgan@gmail.com>
Date: Fri, 3 Feb 2012 12:10:23 -0800
Subject: [PATCH 21/47] Fixed mislabeled dependencies for Fedora 15

---
 cmake/Packaging/Fedora-15.cmake |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/cmake/Packaging/Fedora-15.cmake b/cmake/Packaging/Fedora-15.cmake
index 1ec017a..278d689 100644
--- a/cmake/Packaging/Fedora-15.cmake
+++ b/cmake/Packaging/Fedora-15.cmake
@@ -1,10 +1,10 @@
 SET(PACKAGE_DEPENDS_GRUEL_RUNTIME "boost-python" "glibc")
 SET(PACKAGE_DEPENDS_GRUEL_PYTHON "python")
 SET(PACKAGE_DEPENDS_CORE_RUNTIME "fftw-libs" "gsl")
-SET(PACKAGE_DEPENDS_QTGUI_RUNTIME "qt" "qwt4")
+SET(PACKAGE_DEPENDS_QTGUI_RUNTIME "qt" "qwt")
 SET(PACKAGE_DEPENDS_QTGUI_PYTHON "PyQt4" "PyQwt")
 SET(PACKAGE_DEPENDS_GRC "python" "numpy" "gtk2" "python-lxml" "python-cheetah")
-SET(PACKAGE_DEPENDS_WXGUI "python-wxgtk2.8" "python" "numpy")
+SET(PACKAGE_DEPENDS_WXGUI "wxGTK" "python" "numpy")
 SET(PACKAGE_DEPENDS_VIDEO_SDL_RUNTIME "SDL")
 SET(PACKAGE_DEPENDS_UHD_RUNTIME "uhd")
 SET(PACKAGE_DEPENDS_AUDIO_RUNTIME "pulseaudio" "alsa-lib" "jack-audio-connection-kit")
-- 
1.7.9.1

