From 1567cbc4ca15c3b07f0fe85dd4e1c0ac767ccca0 Mon Sep 17 00:00:00 2001
From: mi-a <kermitalter@gmail.com>
Date: Wed, 23 Apr 2014 11:42:28 +0300
Subject: [PATCH 22/33] docs - pmt

---
 docs/doxygen/other/pmt.dox |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/docs/doxygen/other/pmt.dox b/docs/doxygen/other/pmt.dox
index 04f58aa..4d54461 100644
--- a/docs/doxygen/other/pmt.dox
+++ b/docs/doxygen/other/pmt.dox
@@ -179,8 +179,8 @@ key0 = pmt.intern("int")
 val0 = pmt.from_long(123)
 val1 = pmt.from_long(234)
 
-key1 = pmt.tern("double")
-val2 = pmt.om_double(5.4321)
+key1 = pmt.intern("double")
+val2 = pmt.from_double(5.4321)
 
 # Make an empty dictionary
 a = pmt.make_dict()
-- 
1.7.10.4

