From 0cad02d7f05c60762c8811bc6e05c9f5b49619c8 Mon Sep 17 00:00:00 2001
From: Jaroslav Skarvada <jskarvad@redhat.com>
Date: Tue, 24 Sep 2013 13:31:37 -0400
Subject: [PATCH 30/40] cmake: adding library suffix variable for setting the
 cmake install directory.

---
 CMakeLists.txt |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 57e6474..5a56172 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -312,7 +312,7 @@ SET(cmake_configs
 )
 
 if(NOT CMAKE_MODULES_DIR)
-   set(CMAKE_MODULES_DIR lib/cmake/gnuradio)
+   set(CMAKE_MODULES_DIR lib${LIB_SUFFIX}/cmake/gnuradio)
 endif(NOT CMAKE_MODULES_DIR)
 
 # Install all other cmake files into same directory
-- 
1.7.10.4

