From f5e33003152c693a87f96cb92b7f5a98b627f3b2 Mon Sep 17 00:00:00 2001
From: Nicholas Corgan <nick.corgan@ettus.com>
Date: Mon, 6 Feb 2012 15:24:37 -0800
Subject: [PATCH 30/47] CMake package uses "X.X.X" version number instead of
 "vX.X.X"

---
 cmake/Modules/GrPackage.cmake |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/cmake/Modules/GrPackage.cmake b/cmake/Modules/GrPackage.cmake
index 65fbcc3..a65a2f8 100644
--- a/cmake/Modules/GrPackage.cmake
+++ b/cmake/Modules/GrPackage.cmake
@@ -103,7 +103,7 @@ endfunction(CPACK_COMPONENT)
 set(CPACK_PACKAGE_DESCRIPTION_SUMMARY "GNU Radio - The GNU Software Radio")
 set(CPACK_PACKAGE_VENDOR              "Free Software Foundation, Inc.")
 set(CPACK_PACKAGE_CONTACT             "Discuss GNURadio <discuss-gnuradio@gnu.org>")
-string(SUBSTRING "${VERSION}" 1 -1 CPACK_PACKAGE_VERSION)
+string(REPLACE "v" "" CPACK_PACKAGE_VERSION ${VERSION})
 set(CPACK_RESOURCE_FILE_LICENSE ${CMAKE_SOURCE_DIR}/README)
 set(CPACK_RESOURCE_FILE_README ${CMAKE_SOURCE_DIR}/README)
 set(CPACK_RESOURCE_FILE_WELCOME ${CMAKE_SOURCE_DIR}/README)
-- 
1.7.9.1

