From 38a146d4b6238a161ce8567ab62467ced79d0571 Mon Sep 17 00:00:00 2001
From: gnieboer <gnieboer@corpcomm.net>
Date: Tue, 13 Jun 2017 21:02:13 -0400
Subject: [PATCH 33/58] gr-fft: Adds w32 specific headers to fft.cc

---
 gr-fft/lib/fft.cc | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/gr-fft/lib/fft.cc b/gr-fft/lib/fft.cc
index ef81583187..37390a69bb 100644
--- a/gr-fft/lib/fft.cc
+++ b/gr-fft/lib/fft.cc
@@ -36,6 +36,10 @@ static int my_fftw_read_char(void *f) { return fgetc((FILE *) f); }
 #define fftw_import_wisdom_from_file(f) fftw_import_wisdom(my_fftw_read_char, (void*) (f))
 #define fftwf_import_wisdom_from_file(f) fftwf_import_wisdom(my_fftw_read_char, (void*) (f))
 #define fftwl_import_wisdom_from_file(f) fftwl_import_wisdom(my_fftw_read_char, (void*) (f))
+#include <fcntl.h> 
+#include <io.h>
+#define O_NOCTTY 0
+#define O_NONBLOCK 0
 #endif //_MSC_VER
 
 #include <stdlib.h>
-- 
2.11.0

