From 97059450f485bb1bf74f5dc2b2bf8b1852a0a7bf Mon Sep 17 00:00:00 2001
From: Johnathan Corgan <johnathan@corganlabs.com>
Date: Mon, 30 May 2016 11:34:02 -0700
Subject: [PATCH 39/46] volk: updated to current maint 78c8bc4

---
 volk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/volk b/volk
index f1c4b55..78c8bc4 160000
--- a/volk
+++ b/volk
@@ -1 +1 @@
-Subproject commit f1c4b5534db875a05e71b3b0dd3fbf486ef2c6dc
+Subproject commit 78c8bc4a0e49487e1f787f0fe77d8c3dace7c7fa
-- 
2.1.4

