From 95ddbdd12baadcb327eaf17ccdc75657f2f99cda Mon Sep 17 00:00:00 2001
From: Tom Rondeau <trondeau@vt.edu>
Date: Fri, 24 Feb 2012 15:57:45 -0500
Subject: [PATCH 45/47] digital: forgot to include new xml in Makefile.am.

---
 gr-digital/grc/Makefile.am |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/gr-digital/grc/Makefile.am b/gr-digital/grc/Makefile.am
index b1ebba4..e6dd2aa 100644
--- a/gr-digital/grc/Makefile.am
+++ b/gr-digital/grc/Makefile.am
@@ -28,6 +28,7 @@ dist_grcblocks_DATA = \
 	digital_binary_slicer_fb.xml		\
 	digital_clock_recovery_mm_xx.xml	\
 	digital_constellation_decoder_cb.xml	\
+	digital_constellation_receiver_cb.xml	\
 	digital_correlate_access_code_bb.xml	\
 	digital_costas_loop_cc.xml		\
 	digital_cma_equalizer_cc.xml		\
-- 
1.7.9.1

